internal/rule/condition.go

```diff
@@ -11,74 +12,92 @@ type EscalationFilter struct {
 	IncidentSeverity event.Severity
 }
 
-func (c *EscalationFilter) EvalEqual(key string, value string) (bool, error) {
+func (e *EscalationFilter) ExtractRetryAfter(conditions []filter.Condition, incidentAge time.Duration) time.Duration {
+	var retryAfter time.Duration
+	for _, condition := range conditions {
+		if condition.Column == "incident_age" {
+			age, err := time.ParseDuration(condition.Value)
+			if err != nil {
+				continue
+			}
+
+			if incidentAge < age && (retryAfter <= 0 || retryAfter > age) {
+				retryAfter = age
+			}
+		}
+	}
+
+	return retryAfter
+}
```

Comment:
This function is a really good candidate for a doc comment explaining what it does.